### ISSUE TITLE
CreateTableOne drops unused factor levels

### ISSUE THREAD

**Issue by DavisBrian** [kaz-yos/tableone#27]
It seams CreateTableOne relevels a data frame with factor variables and drops unused levels.

````r
library(dplyr)
library(tableone)

tribble(
  ~Group, ~value,
  #------|-----
  "G1"   , "Level 1" ,
  "G1"   , "Level 2" ,
  "G1"   , "Level 2" ,
  "G1"   , "Level 3" ,
  "G2"   , "Level 3" ,
  "G2"   , "Level 3" 
)%>%
  mutate(value = as.factor(value)) -> mydf

t1 <- CreateTableOne(vars = "value", factorVars = "value", strata = "Group", data = mydf)
print(t1, showAllLevels = TRUE)
````
This gives me what I'd expect
````
           Stratified by Group
            level   G1        G2         p      test
  n                 4         2                     
  value (%) Level 1 1 (25.0)  0 (  0.0)   0.223     
            Level 2 2 (50.0)  0 (  0.0)             
            Level 3 1 (25.0)  2 (100.0)             
````
Now if I subset out the first row of my data frame
````
x <- mydf[-1, ]

str(x)
````
I still have 3 levels as expected

````
Classes ‘tbl_df’, ‘tbl’ and 'data.frame':	5 obs. of  2 variables:
 $ Group: chr  "G1" "G1" "G1" "G2" ...
 $ value: Factor w/ 3 levels "Level 1","Level 2",..: 2 2 3 3 3

````
````
t2 <- CreateTableOne(vars = "value", factorVars = "value", strata = "Group", data = x)
print(t2, showAllLevels = TRUE)
````

But now I'm only shown 2 of my 3 levels.  I would have expected Level 1 to be in the output but 0 (0.) for both groups.

````
           Stratified by Group
            level   G1        G2         p      test
  n                 3         2                     
  value (%) Level 2 2 (66.7)  0 (  0.0)   0.576     
            Level 3 1 (33.3)  2 (100.0)   
````

Looking at the structure of the tableone object it appears the data is releveled.

````
> str(t2)
List of 3
 $ ContTable: NULL
 $ CatTable :List of 2
  ..$ G1:List of 1
  .. ..$ value:'data.frame':	2 obs. of  7 variables:
  .. .. ..$ n          : int [1:2] 3 3
  .. .. ..$ miss       : int [1:2] 0 0
  .. .. ..$ p.miss     : num [1:2] 0 0
  .. .. ..$ level      : Factor w/ 2 levels "Level 2","Level 3": 1 2
  .. .. ..$ freq       : 'table' int [1:2(1d)] 2 1
  .. .. .. ..- attr(*, "dimnames")=List of 1
  .. .. .. .. ..$ x: chr [1:2] "Level 2" "Level 3"
  .. .. ..$ percent    : table [1:2(1d)] 66.7 33.3
  .. .. .. ..- attr(*, "dimnames")=List of 1
  .. .. .. .. ..$ x: chr [1:2] "Level 2" "Level 3"
````

Is there a way to keep the original levels?  This issue comes up for me quite a bit where'd I 'd like to show all possible levels but some are quite rare (like death) in my actual data set.

**Comment by kaz-yos** [comment]
````r
library(dplyr)
library(tableone)

tribble(
  ~Group, ~value,
  #------|-----
  "G1"   , "Level 1" ,
  "G1"   , "Level 2" ,
  "G1"   , "Level 2" ,
  "G1"   , "Level 3" ,
  "G2"   , "Level 3" ,
  "G2"   , "Level 3" 
)%>%
  mutate(value = as.factor(value)) -> mydf

t1 <- CreateTableOne(vars = "value", 
                                     ## factorVars = "value", 
                                     strata = "Group", data = mydf)
print(t1, showAllLevels = TRUE)
````

should probably do what you want.

**Comment by DavisBrian** [comment]
Worked perfectly on my real data set.  Thanks for the quick response and excellent package.

**Comment by Sirhurryup** [comment]
I created an object of categorical variables that were of character type in my dataset. Then I created a Tableone object using the tutorial convention. When I execute that file, the categorical variables never return. I tried to convert my character variables to factors and run the CreateTableOne() to produce the output. Still, the categorical variables never appear. 

`require(tableone)`
`dput((names(demo)`

`
con_vars <- c("days_los_estimate", 
              "days_hemodialysis", "days_peritoneal_dialysis",
             "days_crrt","days_vent","age_estimate", 
             "age_exact", "ed_los_in_hrs", "ed_ip_los_in_hrs")`

`catvars <- c("hemodialysis", "peritoneal_dialysis", "crrt", 
             "vent","dialysis_any","race", "sex",
             "planning_payor_category_2010",  
             "death_indicator", "age_65")`

`sep <- CreateTableOne(vars = convars, data = demo,
                       factorVars = catvars)`

`sep`

                                       Overall        
  n                                      2535         
  days_los_estimate (mean (SD))         15.51 (27.41) 
  days_hemodialysis (mean (SD))          0.68 (4.32)  
  days_peritoneal_dialysis (mean (SD))   0.04 (0.63)  
  days_crrt (mean (SD))                  0.38 (2.37)  
  days_vent (mean (SD))                  2.46 (8.69)  
  age_estimate (mean (SD))              60.83 (16.20) 
  age_exact (mean (SD))                 61.33 (16.20) 
  ed_los_in_hrs (mean (SD))              9.68 (15.86) 
  ed_ip_los_in_hrs (mean (SD))         440.98 (794.40)

Do the variables that require transformation need to be in another format? Unsure of how to display the categorical variables. I tried using as.factor in a mutate() to change the data type and the tableone object still only returns the continuous variables.
